### ISSUE TITLE
Is there a way to check if the video settings are really sent on the callee?

### ISSUE THREAD

**Issue by TerenceAnton** [EricssonResearch/openwebrtc#418]
Hi, I am currently using the Native app development guide for iOS to make a WebRTC app. I am tinkering with the OpenWebRTCSettings class. I was wondering how can I check if the video settings I set will be rendered on the receiving end of the call. As an example:

settings.videoFramerate = 30.0
settings.videoWidth = 640
settings.videoHeight = 480

I am testing with an iPad mini as the caller and the Chrome browser as the callee. I tried to check in chrome://webrtc-internals for this but there is no information regarding this.

**Comment by superdump** [comment]
I think the width, height, framerate are all printed in Chrome's WebRTC internals. They're displayed as graphs in the stats part. If you scroll down to where there are the send and recv stats and expand the recv stats sections, you can find the one that is a video stream and you should see something like this:

![screen shot 2015-06-29 at 11 47 49](https://cloud.githubusercontent.com/assets/302146/8405446/d498823e-1e54-11e5-817d-ba987d3dee53.png)

Note googFrameHeightReceived, googFrameWidthReceived and googFrameRateReceived.

**Comment by TerenceAnton** [comment]
Hi Robert,

Thank you very much for the reply. Here is a screenshot of chrome://webrtc-internals. I used the NativeDemo app for iOS and demo.openwebrtc.io:38080 for the browser. 
![chrome](https://cloud.githubusercontent.com/assets/4739606/8406595/74ec88d0-1e90-11e5-9d99-97a8fd5b5390.PNG)

Unfortunately, there was no stats for recv. I am able to see changes on selfView when I tweak the video resolution but I am not sure if the app is really sending that settings to the browser. Any thoughts on this?

**Comment by superdump** [comment]
Is your Chrome client actually receiving any video?

**Comment by TerenceAnton** [comment]
Yes Robert, I am able to see the local feed on my iPad to the remoteFeed of
the browser.

**Comment by superdump** [comment]
Then perhaps you're looking at the wrong session in chrome://webrtc-internals? You can switch between them at the top.

![screen shot 2015-06-29 at 13 49 14](https://cloud.githubusercontent.com/assets/302146/8407312/be081b22-1e65-11e5-8020-d60d38f59b06.png)

**Comment by TerenceAnton** [comment]
Here is the screenshot of chrome://webrtc-internals:
![chrome2](https://cloud.githubusercontent.com/assets/4739606/8407470/5a184bbc-1e99-11e5-86ef-a4cae44de86a.PNG)

Unfortunately, I am still not able to see stats for recv. I only have one session and I can only see the sent statistics.

**Comment by superdump** [comment]
Did you scroll down to the bottom of the list on the left?

**Comment by TerenceAnton** [comment]
Yes, there was no recv stat. This is weird because if I run the demo
website with 2 browsers there is a recv stat. Are you able to reproduce in
your side?

**Comment by superdump** [comment]
Sorry for the seemingly stupid questions, it's just a very odd situation that Chrome is receiving and displaying video but it has no stats for it. Do you see video from Chrome on the other side?

**Comment by TerenceAnton** [comment]
It is alright Robert. Yes, I am able to see the video from Chrome on the
other side

**Comment by superdump** [comment]
Which side did you start the call from? Is it the same if you start the call from the other side?

**Comment by TerenceAnton** [comment]
I tried both. I called from the iOS app and there was no recv stat. And then I called from the browser and still there was no recv stat. 

I thought that this might be a problem for iOS devices so I installed another app, appear.in. I was able to get the recv stat. Here is a screenshot:
![chrome3](https://cloud.githubusercontent.com/assets/4739606/8408164/46e25a72-1ea0-11e5-939c-7e2c4f3c507f.PNG)

**Comment by superdump** [comment]
Odd. OK, thanks for the report. We will have to investigate this. For our clarity, can you make note here of the iOS, OpenWebRTC and Chrome software versions and the device you're using?

**Comment by TerenceAnton** [comment]
iPad Mini (A1489) = iOS 8.3
Chrome on Windows 7 = 43.0.2357.130
Chrome on OSX Yosemite 10.10.3 = 43.0.2357.130

I got the NativeDemo app from https://github.com/EricssonResearch/openwebrtc-examples. I tried to experiment on the iOS SDK. I followed the guide Native app development guide https://github.com/EricssonResearch/openwebrtc-examples/wiki/Developing-a-native-iOS-app and used the SDK on https://github.com/EricssonResearch/openwebrtc-ios-sdk. Please let me know if you need more information. Thanks for the help Robert.
